### PR TITLE
Fixes the issue of storage unmounting during odo push

### LINKED ISSUE #1539: Storage unmounting during odo push
```
odo create nodejs
odo push 
odo storage create <name> --path <path> --size <size>
odo storage list
odo push
odo storage list
```

The storage gets unmounted during the push.

### PR DESCRIPTION
**Pull request by mik-dass**
fixes #1539

### CODE REVIEW

File: pkg/occlient/occlient.go [hunk 1]
```diff
@@ -1287,6 +1287,9 @@ func (c *Client) PatchCurrentDC(name string, dc appsv1.DeploymentConfig, prePatc
 		}
 	}
 
+	// copy the any remaining volumes and volume mounts
+	copyVolumesAndVolumeMounts(dc, currentDC, existingCmpContainer)
```
Comment (kadel) [inline pkg/occlient/occlient.go]: Why is this called again? I is already inside the `prePatchDCHandler` that is called a few lines before this.

Comment (mik-dass) [reply pkg/occlient/occlient.go]: `prePatchDCHandler()` above is only called during local to git updates

Comment (kadel) [reply pkg/occlient/occlient.go]: Wouldn't it be better to remove it from prePatchHandler and call just call it here?

**Comment by mik-dass** [comment]
@kadel Done

**Comment by kadel** [comment]
ping @anmolbabu @girishramnani @surajnarwade

**Comment by cdrage** [comment]
@mik-dass the code looks good, however, are you able to add a test (even e2e?)

**Comment by amitkrout** [comment]
> > Copied from issue description
> > ```
> > odo create nodejs
> > odo push 
> > odo storage create <name> --path <path> --size <size>
> > odo storage list
> > odo push
> > odo storage list
> > ```
> > @mik-dass Such scenario is commented in our existing test due to new odo workflow . Can you please add a test scenario for this.
> 
> I don't think that this issue is supposed to fix this scenario. There is #1547

If this is the case then i will be happy to see the scenario coverage in #1547

**Comment by mik-dass** [comment]
> @mik-dass the code looks good, however, are you able to add a test (even e2e?)

I will add the e2e test as part of #1547 :)

**Pull request merged**